File: monolith/src/main/java/com/itinfo/service/impl/UsersServiceImpl.java
package com.itinfo.service.impl;

import com.itinfo.dao.UsersDao;
import com.itinfo.service.UsersService;
import com.itinfo.service.engine.ConnectionService;
import com.itinfo.model.UserVo;
import com.itinfo.security.SecurityUtils;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;
import java.util.List;

import lombok.extern.slf4j.Slf4j;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
@Slf4j
public class UsersServiceImpl implements UsersService {
	@Autowired private ConnectionService connectionService;
	@Autowired private UsersDao usersDao;
	@Autowired private SecurityUtils securityUtils;

	public List<UserVo> retrieveUsers() {
		log.info("... retrieveUsers");
		return usersDao.retrieveUsers();
	}

	public Integer removeUsers(List<UserVo> users) {
		log.info("... removeUsers");
		return usersDao.removeUsers(users);
	}

	public Boolean isExistUser(UserVo user) {
		log.info("... isExistUser");
		return usersDao.isExistUser(user);
	}

	public Integer addUser(UserVo user) {
		log.info("... addUser");
		user.setPassword(SecurityUtils.createHash(user.getPassword()));
		return usersDao.addUser(user);
	}

	public String login(String id) {
		log.info("... login("+id+")");
		return usersDao.login(id);
	}

	public UserVo retrieveUser(String id) {
		log.info("... retrieveUser("+id+")");
		return usersDao.retrieveUser(id);
	}

	public Integer updateUser(UserVo user) {
		log.info("... retrieveUser");
		return usersDao.updateUser(user);
	}

	public Integer updatePassword(UserVo user) {
		log.info("... updatePassword");
		user.setNewPassword(SecurityUtils.createHash(user.getNewPassword()));
		return usersDao.updatePassword(user);
	}

	public Integer updateLoginCount(UserVo user) {
		log.info("... updateLoginCount");
		return usersDao.updateLoginCount(user);
	}

	public void setBlockTime(UserVo user) {
		log.info("... setBlockTime");
		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMddHHmmss");
		String blockTime = LocalDateTime.now().plusMinutes(10L).format(formatter);
		user.setBlockTime(blockTime);
		usersDao.setBlockTime(user);
	}

	public void initLoginCount(String userId) {
		log.info("... initLoginCount("+userId+")");
		usersDao.initLoginCount(userId);
	}
}

File: monolith/src/main/java/com/itinfo/service/impl/UsersServiceImpl.kt
package com.itinfo.service.impl

import com.itinfo.common.LoggerDelegate
import com.itinfo.dao.OvirtUsersDao
import com.itinfo.model.UserVo
import com.itinfo.security.createHash
import com.itinfo.service.UsersService

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

import java.time.LocalDateTime
import java.time.format.DateTimeFormatter


/**
 * [UsersServiceImpl]
 * 사용자 관리 서비스 응용
 *
 * @author chlee
 * @since 2023.12.07
 */
@Service
class UsersServiceImpl : UsersService {
    @Autowired private lateinit var usersDao: OvirtUsersDao

    /**
     * [UsersService.fetchUsers]
     */
    override fun fetchUsers(): List<UserVo> {
        log.info("... retrieveUsers")
        val users: List<UserVo> =
            usersDao.retrieveUsers().also { log.debug("found: $it") }
        return users
    }

    /**
     * [UsersService.removeUsers]
     */
    override fun removeUsers(users: List<UserVo>): Int {
        log.info("... removeUsers")
        return usersDao.removeUsers(users)
    }

    /**
     * [UsersService.isExistUser]
     */
    override fun isExistUser(user: UserVo): Boolean {
        log.info("... isExistUser")
        return usersDao.isExistUser(user)
    }

    /**
     * [UsersService.addUser]
     */
    override fun addUser(user: UserVo): Int {
        log.info("... addUser")
        user.password = user.password.createHash()
        return usersDao.addUser(user)
    }

    override fun login(id: String): String {
        log.info("... login($id)")
        return usersDao.login(id)
    }

    override fun fetchUser(id: String): UserVo? {
        log.info("... retrieveUser($id)")
        return usersDao.retrieveUser(id)
    }

    override fun updateUser(user: UserVo): Int {
        log.info("... retrieveUser")
        return usersDao.updateUser(user)
    }

    override fun updatePassword(user: UserVo): Int {
        log.info("... updatePassword")
        user.newPassword = user.newPassword.createHash()
        return usersDao.updatePassword(user)
    }

    override fun updateLoginCount(user: UserVo): Int {
        log.info("... updateLoginCount")
        return usersDao.updateLoginCount(user)
    }

    override fun setBlockTime(user: UserVo) {
        log.info("... setBlockTime")
        user.blockTime = LocalDateTime.now()
                .plusMinutes(10L)
                .format(DateTimeFormatter.ofPattern("yyyyMMddHHmmss"))
        usersDao.setBlockTime(user)
    }

    override fun initLoginCount(userId: String) {
        log.info("... initLoginCount($userId)")
        usersDao.initLoginCount(userId)
    }
    
    companion object {
        private val log by LoggerDelegate()
    }
}
